File: src/main/java/org/jdc/template/domain/attached/crossdatabasequery/CrossDatabaseQuery.java
/*
 * CrossDatabaseQuery.java
 *
 * Created: 09/18/2014 09:40:57
 */



package org.jdc.template.domain.attached.crossdatabasequery;

import android.content.ContentValues;
import android.database.Cursor;

import org.jdc.template.domain.main.individual.IndividualConst;

public class CrossDatabaseQuery extends CrossDatabaseQueryBaseRecord {

    public static final String QUERY = "(" +
            "SELECT " +
            IndividualConst.FULL_C_ID + " AS " + CrossDatabaseQueryConst.C_ID + ", " +
            IndividualConst.FULL_C_FIRST_NAME + " AS " + CrossDatabaseQueryConst.C_NAME + ", " +
            " 1 " + " AS " + CrossDatabaseQueryConst.C_TYPE +
            " FROM " + IndividualConst.TABLE +
            ")";
    public static final String QUERY_RAW = "SELECT * FROM " + QUERY;

    public CrossDatabaseQuery(Cursor cursor) {
        setContent(cursor);
    }

    public CrossDatabaseQuery(ContentValues values) {
        setContent(values);
    }

    public CrossDatabaseQuery() {
    }


}

File: src/main/kotlin/org/jdc/template/domain/attached/crossdatabasequery/CrossDatabaseQuery.kt
/*
 * CrossDatabaseQuery.kt
 *
 * Created: 02/14/2016 07:02:10
 */



package org.jdc.template.domain.attached.crossdatabasequery

import android.content.ContentValues
import android.database.Cursor
import org.dbtools.query.sql.SQLQueryBuilder

class CrossDatabaseQuery : CrossDatabaseQueryBaseRecord {

    companion object {
        val QUERY: String
        init {
            QUERY = SQLQueryBuilder()
                    .field(CrossDatabaseQueryConst.FULL_C_ID, CrossDatabaseQueryConst.C_ID)
                    .field(CrossDatabaseQueryConst.FULL_C_NAME, CrossDatabaseQueryConst.C_NAME)
                    .field(CrossDatabaseQueryConst.FULL_C_TYPE, CrossDatabaseQueryConst.C_TYPE)
                    .table("FROM SOME TABLE(S)")
                    .buildQuery();
        }
    }

    constructor() {
    }

    constructor(cursor: Cursor) {
        setContent(cursor)
    }

    constructor(values: ContentValues) {
        setContent(values)
    }


}